packages/assets-controllers/src/NftController.ts

```diff
@@ -512,6 +512,53 @@ export class NftController extends BaseController<
     });
   }
 
+  /**
+   * Fetches NFT Collection Metadata from the NFT API.
+   *
+   * @param contractAddresses - The contract addresses of the NFTs.
+   * @param chainId - The chain ID of the network where the NFT is located.
+   * @returns NFT collections metadata.
+   */
+  async fetchNftCollectionsMetadata(
+    contractAddresses: string[],
+    chainId: Hex,
+  ): Promise<{
+    collections: Collection[];
+  }> {
+    return await this.#getNftCollectionInformationFromApi(
+      contractAddresses,
+      chainId,
+    );
+  }
+
+  async #getNftCollectionInformationFromApi(
+    contractAddress: string[],
+    chainId: Hex,
+  ) {
+    const urlParams = new URLSearchParams({
+      chainId,
+    });
+
+    for (const address of contractAddress) {
+      urlParams.append('contract', address);
+    }
+
+    return await handleFetch(
+      `${this.#getNftCollectionApi()}?${urlParams.toString()}`,
+      {
+        headers: {
+          Version: '1',
+        },
+      },
+    );
+  }
+
+  #getNftCollectionApi(): string {
+    // TODO: Either fix this lint violation or explain why it's necessary to ignore.
+    // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+    return `${NFT_API_BASE_URL}/collections`;
```

Comment:
Just for audit sake, can I confirm this is using the reservoir API which includes the `isSpam` property?

Reply:
Yes I have checked it before, and it does included in the response.